ci: add rustfmt check to formatting workflow

lucienfer opened the pull request:
## What does this PR do?

This PR extends the existing `format` job in `.github/workflows/build.yml` to
**check Rust formatting with nightly `rustfmt`**:

```yaml
- name: Check Rust formatting (rustfmt --check)
  run: cargo fmt --all -- --check
  ```

## Why?

•	The repository already defines a rustfmt.toml with nightly-only settings (imports_granularity, use_small_heuristics = "Max", etc.).
•	The CI installs nightly rustfmt but wasn’t actually using it.
•	Currently, dprint/check verifies formatting for Markdown/PO/JS/etc., but not Rust code.
This PR makes sure PRs fail if Rust code is not formatted according to rustfmt.toml.

## How was it tested?

•	Ran locally with:
```bash
rustup default nightly
rustup component add rustfmt
cargo fmt --all -- --check
dprint fmt --check
```
Both checks pass on a clean branch.

•	CI workflow updated accordingly.
If Rust code is unformatted, the format job will now fail.

**Closes #2794**

mgeisler commented:
Hey @lucienfer, thanks for the PR! I'm surprised this is necessary: `dprint` is supposed to run `rustfmt` as well, but I guess it's somehow not doing this (since you include formatting changes in this PR)?

lucienfer commented:
You’re right, after digging a bit I realized that dprint wasn’t actually checking the Rust formatting properly because it was configured to run rustfmt on the stable toolchain. Since the repo’s `rustfmt.toml` uses unstable options like `imports_granularity = "module"`, those settings were simply being ignored.

To fix this I made two changes:
	1.	I added the `unstable_features = true` flag in `rustfmt.toml` to make sure nightly-specific settings are explicitly enabled.
	2.	Instead of relying on `cargo fmt` in CI, I updated the `dprint.json` config so that it directly runs `rustup run nightly rustfmt`. This way dprint is aligned with the nightly toolchain already used in CI, and it will fail if the code isn’t correctly formatted.

This should ensure that formatting is consistently checked both locally (via dprint) and in CI.

mgeisler commented on xtask/src/main.rs:
```diff
@@ -91,6 +94,30 @@ fn execute_task() -> Result<()> {
+    const PINNED_NIGHTLY: &str = "nightly-2025-09-01";
+
+    let rustup_steps = [
+        // Installe la toolchain sans les composants inutiles
```
Heh, let's write this in English :smile:

mgeisler commented on CONTRIBUTING.md:
````diff
@@ -53,6 +53,16 @@ this:
+cargo xtask install-tools
+```
+
+This will install the pinned nightly toolchain (`nightly-2025-09-01`) and add
````
Let's avoid mentioning the precise version here since we'll surely forget to update this later :smile:

lucienfer replied:
I had even forgotten that it needed to be updated 😂

mgeisler commented on xtask/src/main.rs:
```diff
@@ -91,6 +94,30 @@ fn execute_task() -> Result<()> {
+        let status = std::process::Command::new("rustup")
+            .args(args)
+            .status()
+            .expect("Failed to execute rustup");
```
Please don't use `expect` (or `unwrap`), instead return the `Result` back with `?`. @egithinji, can you review this part? We should (eventually) have some useful patterns in the `xtask` code to easily and quickly execute shell commands.

mgeisler replied:
Ah, I see! Let's update the existing code in a different PR to keep this focused on your changes.
